core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -79,6 +79,23 @@ public static CloseableIterable<FileScanTask> splitFiles(
     return CloseableIterable.combine(splitTasks, tasks);
   }
 
+  public static <T extends ScanTask> CloseableIterable<T> splitScanTasks(
```

Comment:
Instead of doing this, we should offer another method with the following signature.

```
public static <T extends ScanTask> List<ScanTaskGroup<T>> planTaskGroups(
    List<T> tasks,
    long splitSize,
    int lookback,
    long openFileCost) {
  ...
}
```

This will work in the scan for staged tasks as well as for `SparkPartitioningAwareScan`. Under the hood, it will simply delegate to the existing `planTaskGroups` but will wrap list into `CloseableIterable`.

Reply:
Done